packages/astro/src/core/cookies/cookies.ts

```diff
@@ -148,6 +150,13 @@ class AstroCookies implements AstroCookiesInterface {
 	 * @param options Options for the cookie, such as the path and security settings.
 	 */
 	set(key: string, value: string | Record<string, any>, options?: AstroCookieSetOptions): void {
+		if (this.#consumed) {
+			console.error(new Error(
+				"Astro.cookies.set() was called after the cookies have already been sent to the browser.\n" +
+				"This may have happened if this method was called in an imported component.\n" +
+				"Please make sure that Astro.cookies.set() is only called in the frontmatter of the main page."
+			));
+		}
```

Comment:
![image](https://github.com/withastro/astro/assets/69170106/bb7a59af-1838-45a5-8f33-3d37afa77ced)

Reply:
```suggestion
		if (this.#consumed) {
			console.error(new Error(
				"Astro.cookies.set() was called after the cookies were already sent to the browser.\n" +
				"This may have happened if this method was called in an imported component.\n" +
				"Please make sure that Astro.cookies.set() is only called in the frontmatter of the main page."
			));
		}
```